api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -44,6 +44,14 @@ message DiscoveryRequest {
   // which will be explicitly enumerated in resource_names.
   repeated string resource_names = 3;
 
+  // An alternative to resource_names that specifies an optional subscription
+  // ID for each resource locator subscribed to. Resources that match
+  // each resource locator will be tagged with the associated subscription
+  // ID in the Resource.new_subscription_ids field in the response.
+  // Note that the client must be prepared for each resource in the
+  // response to be wrapped in a Resource message.
+  map<string, uint64> resource_locators = 7;
```

Comment:
Nit: maybe prefer explicit repeated message on this. Given the non-determinism of map representation, it sure makes debugging awkward.